AudioPlayer/AudioPlayer/player/AudioPlayerState.swift

```diff
@@ -25,13 +25,13 @@ public enum AudioPlayerError: Error {
 /// - stopped: The player is stopped.
 /// - waitingForConnection: The player is waiting for internet connection.
 /// - failed: An error occured. It contains AVPlayer's error if any.
-public enum AudioPlayerState {
-    case buffering
-    case playing
-    case paused
-    case stopped
-    case waitingForConnection
-    case failed(AudioPlayerError)
+@objc public enum AudioPlayerState: Int {
```

Comment:
Maybe we should just revert to the current AudioPlayerState, then have a `didSet` func on the variable which mirrors the state to an obj-c object, which encapsulates state and failedError.
This would also require a separate `objcDelegate`, perhaps with some smart way to forward swift delegate callbacks to it.

At least then we won't be breaking or gimping the Swift version of AudioPlayer. It's also not so much fun having forego all the nice Swift features for the sake of Obj-c compatibility.